src/Network/Network/VirtualRouter/NewAzureRMVirtualRouterCommand.cs

```diff
@@ -0,0 +1,159 @@
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable which represents resource tags.",
+            ValueFromPipelineByPropertyName = true)]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Do not ask for confirmation if you want to overwrite a resource")]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            var present = true;
+            try
+            {
+                this.NetworkClient.NetworkManagementClient.VirtualRouters.Get(this.ResourceGroupName, this.Name);
+            }
+            catch (Exception ex)
+            {
+                if (ex is Microsoft.Azure.Management.Network.Models.ErrorException || ex is Rest.Azure.CloudException)
+                {
+                    // Resource is not present
+                    present = false;
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            if (present)
+            {
+                throw new PSArgumentException(string.Format(Properties.Resources.ResourceAlreadyPresentInResourceGroup, this.Name, this.ResourceGroupName));
+            }
+
+            string hostedGatewayId = null;
+
+            //// Resolve the virtual wan
+            if (ParameterSetName.Equals(VirtualRouterParameterSetNames.ByHostedGateway, StringComparison.OrdinalIgnoreCase))
+            {
+                hostedGatewayId = this.HostedGateway.Id;
+            }
+            else if (ParameterSetName.Equals(VirtualRouterParameterSetNames.ByHostedGatewayId, StringComparison.OrdinalIgnoreCase))
+            {
+                hostedGatewayId = this.HostedGatewayId;
+            }
+
+            if (string.IsNullOrWhiteSpace(hostedGatewayId))
+            {
+                throw new PSArgumentException(Properties.Resources.VirtualGatewayRequiredForVirtualRouter);
+            }
+
+
+            ConfirmAction(
+                Properties.Resources.CreatingResourceMessage,
+                Name,
+                () =>
+                {
+                    WriteVerbose(String.Format(Properties.Resources.CreatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
+                    PSVirtualRouter virtualRouter = new PSVirtualRouter
+                    {
+                        ResourceGroupName = this.ResourceGroupName,
+                        Name = this.Name,
+                        HostedGateway = new PSResourceId() { Id = hostedGatewayId },
+                        Location = this.Location,
+                        VirtualRouterAsn = 65515
```

Comment:
What does 65515 mean? Can it be move into a constant variable with a comment?

Reply:
Defined this constant in VirtualRouterBaseCmdlet